Userland/Libraries/LibJS/Runtime/VM.cpp

```diff
@@ -989,7 +989,7 @@ void VM::finish_dynamic_import(ScriptOrModule referencing_script_or_module, Modu
     dbgln_if(JS_MODULE_DEBUG, "[JS MODULE] finish_dynamic_import on {}", module_request.module_specifier);
 
     // 1. Let fulfilledClosure be a new Abstract Closure with parameters (result) that captures referencingScriptOrModule, specifier, and promiseCapability and performs the following steps when called:
-    auto fulfilled_closure = [referencing_script_or_module, module_request = move(module_request), promise_capability](VM& vm, GlobalObject& global_object) -> ThrowCompletionOr<Value> {
+    auto fulfilled_closure = [referencing_script_or_module, module_request = move(module_request), resolve_function = make_handle(promise_capability.resolve), reject_function = make_handle(promise_capability.reject)](VM& vm, GlobalObject& global_object) -> ThrowCompletionOr<Value> {
```

Comment:
Is this actually needed? `promise_capability` is on the stack, and that should be visible to the conservative GC scan.

Reply:
It is on the stack here, but these function might only be called later when we are running queued promises. In that case these could be the only references to the promise capability values.

To make it concrete without this `test-js -g` fails.

Reply:
Ah, I see! :)